android-sdk/README.md

```diff
@@ -0,0 +1,93 @@
+# Android Core SDK + Auth, REST, and Sync Service modules Proposal 
+This document outlines the features, architecture choices, tooling needs, and processes for developing the Android SDKs for communicating with services connected by Mobile-Core.  It is hoped that This can also be a model for other platforms as we move forward.
+
+## Goals
+
+### Primary Goal
+ Provide a starting point for development of the Android SDKs based on the general architecture described in https://github.com/aerogear/proposals/blob/1b79c6323997a0fa4e9794bb2b91145702cf5a48/sdk/overview.md as well as define for Android what tools we will build (ie build plugins, IDE tooling, scripts, etc), the relationship between the "Core" SDK and "Service" SDKs, release and distribution processes, and processes for ensuring code quality (dependency management, static analysis, CI/CD, etc)
+
+## Terms
+### Core SDK
+The base library that provides the mechanisms for setting up mobile-core services.  It will include interfaces for Requests, Responses, Services, Modules, and mechanisms for how they become composed.
+### Service Module
+A Service Module connects to a OpenShift service that provides some function.  Examples of services are sync, keycloak, push, rest with RHOAR runtimes, etc
+### Application Module
+This is the code which is written by an application developer to bridge our modules and their application. (for example, application specific business logic)
```

Comment:
Can we assume that modules are internal part of our sdk and users will just import that?
We should handle scenarios when sdk can be added to the existing application.

Reply:
I'm not sure what you mean by "internal".  

The way I see my proposal working is that there is the app and code that calls our modules.  Some of that code may use the SDK to register itself as a module (ie if the developer has a custom piece of code that they want managed by core).  This is what an application module would be.

Service modules are provided by us and integrated with the core sdk already.  They aren't supposed to be used directly by the developer, but can expose some some apis if it is appropriate.

It is like in firebase how there are multiple modules you import in build.gradle, but they are exposed via a few well known APIs.

Reply:
@secondsun I did not read that properly. After your explanation it's clear for me how this will work.

Reply:
So @secondsun an application module would be an Android Library module that I, as a developer, would have in my Android project, only locally?

Reply:
Because the SDK has a lifecycle for the modules to make sure the services are setup in order.  Your application module would be the code that runs last and depends on all of your services defined in mobile-services.json

Reply:
@josemigallas yes